src/transaction.ts

```diff
@@ -94,8 +94,12 @@ export function assembleTransaction(
 
 function buildContractAuth(auths: string[]): xdr.ContractAuth[] {
   const contractAuths: xdr.ContractAuth[] = [];
-  for (const authStr of auths) {
-    contractAuths.push(xdr.ContractAuth.fromXDR(authStr, "base64"));
+  
+  if (auths) {
+    for (const authStr of auths) {
+      contractAuths.push(xdr.ContractAuth.fromXDR(authStr, "base64"));
+    }
   }
+
   return contractAuths;
```

Comment:
This whole thing can be one line, I think:

```suggestion
function buildContractAuth(auths?: string[]): xdr.ContractAuth[] {
  return auths?.map((auth) => xdr.ContractAuth.fromXDR(auth, "base64"));
}
```

Reply:
Oh hm, I'm not sure how this behaves in the `null` case. I think a `?? []` at the end would fix it.

Reply:
almost, but it changes the return type to `ContractAuth[] | undefined` which wasn't the intent, want to adapt whatever simulation data was present into an array instance, empty or otherwise.